src/core/MOM_barotropic.F90

```diff
@@ -5191,31 +5177,40 @@ subroutine bt_mass_source(h, eta, set_cor, G, GV, CS)
 
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
 
-  !$OMP parallel do default(shared) private(eta_h,h_tot,d_eta)
-  do j=js,je
-    do i=is,ie ; h_tot(i) = h(i,j,1) ; enddo
-    if (GV%Boussinesq) then
-      do i=is,ie ; eta_h(i) = h(i,j,1) - G%bathyT(i,j)*GV%Z_to_H ; enddo
-    else
-      do i=is,ie ; eta_h(i) = h(i,j,1) ; enddo
-    endif
-    do k=2,nz ; do i=is,ie
-      eta_h(i) = eta_h(i) + h(i,j,k)
-      h_tot(i) = h_tot(i) + h(i,j,k)
-    enddo ; enddo
+  !$omp target enter data map(alloc: eta_h)
 
-    if (set_cor) then
-      do i=is,ie
-        d_eta = eta_h(i) - eta(i,j)
-        CS%eta_cor(i,j) = d_eta
-      enddo
-    else
-      do i=is,ie
-        d_eta = eta_h(i) - eta(i,j)
-        CS%eta_cor(i,j) = CS%eta_cor(i,j) + d_eta
-      enddo
-    endif
+  ! do concurrent (j=js:je , i=is:ie)
+  !   h_tot(i,j) = h(i,j,1)
+  ! enddo
```

Comment:
h_tot seems to be completely unused, i'm guessing that's why you removed it?  Very strange that it was there though.

I'll ask if there's a story here.